File: miscellaneous/sorting/insertion-sort/20230303/insertion-sort.js
'use strict';

module.exports = (nums) => {
    if (nums.length <= 1) return nums;

    for (let i = 1; i < nums.length; ++i) {
        let j = i - 1;
        const current = nums[i];

        while (j >= 0 && nums[j] > current) {
            nums[j+1] = nums[j--];
        }

        nums[++j] = current;
    }

    return nums;
}

File: miscellaneous/sorting/insertion-sort/20230303/insertion_sort.py
from typing import List


def sort(nums: List[int]) -> List[int]:
    for i in range(len(nums)):
        current = nums[i]
        j = i - 1

        while j >= 0 and nums[j] > current:
            nums[j+1] = nums[j]
            j -= 1

        nums[j+1] = current

    return nums
